### DISCUSSION TITLE
Service running but cannot connect to the UI! (snap)

### DISCUSSION THREAD

**Discussion by jefft4** [zwave-js/zwave-js-ui#2786] [Q&A] [answered]
zwave-js-ui latest, Ubuntu 22, installed via snap package.  I installed zwave-js-ui to replace zwavejs2mqtt, as recommended.

I ported my nodes.json & settings.json from zwavejs2mqtt and changed the service name.  The service is running and I can command my z-wave devices from domoticz via MQTT, but I cannot access the zwjs UI on port 8091.  Netstat shows the node process for zwjs listening:
`Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.1:8091          0.0.0.0:*               LISTEN      19472/node`

and the log says 'Listening on localhost port 8091'.  I need to connect from a remote PC, as the server is headless, so I'm hitting http://192.168.8.200:8091 or http://domoticz:8091 but nothing works, it's always ERR_CONNECTION_REFUSED.

The problem here, I think is that it's listening on 127.0.0.1 not on 0.0.0.0 or the LAN IP of the port (192.168.8.200 / domoticz).

settings.json:
`{
  "mqtt": {
    "name": "zwave-js-ui",
    "host": "localhost",
    "port": 1883,
    "qos": 1,
    "prefix": "zwave",
    "reconnectPeriod": 3000,
    "retain": false,
    "clean": true,
    "auth": false,
    "_ca": "",
    "ca": "",
    "_cert": "",
    "cert": "",
    "_key": "",
    "key": "",
    "disabled": false
  },
  "gateway": {
    "type": 0,
    "plugins": [],
    "authEnabled": false,
    "payloadType": 0,
    "nodeNames": false,
    "hassDiscovery": true,
    "discoveryPrefix": "zwavedisco",
    "logEnabled": true,
    "logLevel": "info",
    "logToFile": true,
    "values": [],
    "includeNodeInfo": true,
    "https": false
  },
  "zwave": {
    "port": "/dev/ttyUSB-zwave-g5",
    "commandsTimeout": 60,
    "logLevel": "info",
    "logEnabled": true,
    "securityKeys": {
      "S0_Legacy": "**hidden**",
      "S2_Authenticated": "**hidden**",
      "S2_Unauthenticated": "**hidden**",
      "S2_AccessControl": "**hidden**"
    },
    "deviceConfigPriorityDir": "/var/snap/zwave-js-ui/current/config",
    "logToFile": true,
    "serverEnabled": false,
    "serverServiceDiscoveryDisabled": false,
    "enableSoftReset": true,
    "enableStatistics": true,
    "serverPort": 3000,
    "maxNodeEventsQueueSize": 100,
    "disclaimerVersion": 1,
    "scales": [],
    "nodeFilter": []
  },
  "backup": {
    "storeBackup": true,
    "storeCron": "0 5 * * *",
    "storeKeep": 21,
    "nvmBackup": true,
    "nvmBackupOnEvent": false,
    "nvmCron": "10 5 * * *",
    "nvmKeep": 21
  }
}`

I tried adding a "host" option under the 'gateway' section and setting it to "domoticz", but that made no difference.

It seems to have the right port defined in the snap config:
`domoticz@domoticz:/var/snap/zwave-js-ui/current$ sudo snap get -d zwave-js-ui
{
        "mqtt": {
                "name": ""
        },
        "ports": {
                "http": 8091
        },
        "server": {
                "force-disable-ssl": false,
                "host": "localhost",
                "port": 8091,
                "ssl": false
        },
        "session": {
                "cookie-secure": "",
                "secret": "49335352-5dfa-11ed-81c9-f70b8cbdfac0"
        }
}`

How do I force it to listen on the correct IP (0.0.0.0)?  All the code and documentation suggests that 0.0.0.0 is the default, so why's mine picking localhost?  Help!

**Comment by jmgiaever** [comment, marked as answer]
Set it to listen to `0.0.0.0`, 

```
sudo snap set zwave-js-ui server.host=0.0.0.0
```

**Comment by jefft4** [comment]
Ah, that's where to force it. Thank you! :-)  I didn't find that anywhere in the documentation.  

The documentation on Github implies that 0.0.0.0 is (reasonably) the default.  Why would it be using 127.0.0.1?  Is that something caused by my system or by the application?  Services normally listen on 0.0.0.0 unless told otherwise; defaulting to localhost causes real pain on smart home systems as most are run headless.

**Comment by jmgiaever** [comment]
There's a help command (`zwave-js-ui.help`) that is snap-specific, but you can always do it manually I'm the settings.json file too.

**Comment by Ronald1817** [comment]
Hi,

I installed snapd 2 days ago on a test pi4 with domoticz and zwave and came accross the hello-world preload error to begin with , the solution to this is comment out the preload rule and then it works , but why does this start with a error , the rest needs to be done but the start is emediately  an error , need to do all the rest, i wonder if there's a step by step plan how to do this with snap , i allso don't want to use docker because it's not really needed and puts overhead on the systeem, it is the  recommended way to go, but why?, have read people who did this, did not get it working.
i do not have a lot zwave devices 9 smoke detectors and few switches and dimmers and that's it,  so migrating should not be an issue.

But want to try this , but with a good step by step plan otherwise one gets lost in all sorts of searching and not reaching a solution .

the features that i've read about the new interface are great , one can update the firmware of the zwave devices if the manufacturer has supplied these files.

Anyone got this working and has wrote down the steps to do this?

**Reply by jmgiaever** [reply]
Hi, Rasbian?

**Comment by Ronald1817** [comment]
Use 👍 
NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=raspbian
ID_LIKE=debian


Am installing Home Assistant on a RPI 4 that is had , and man it's al working out of the box, had a extra AEOTEC wave+ stick , added a few devices and bam it worked with the ZW JS module, am very pleased with that., Domoticz is lagging with features that are on HA out of the box, did not look at HA before but now there's a reason for it.

**Comment by jmgiaever** [comment]
Have a look at the note for Rasbian: https://zwave-js.github.io/zwave-js-ui/#/getting-started/other-methods?id=snap-package

**Comment by Ronald1817** [comment]
thanx, i will look into it.
